bundles/org.openhab.binding.exec/README.md

```diff
@@ -122,34 +121,33 @@ sitemap demo label="Your Value"
 **demo.rules**
 
 ```java
-rule "Your Execution"
-  when
-     Item YourTrigger changed
-  then
-        if(YourTrigger.state == ON){
-                yourcommand_Args.sendCommand("Additional Argument to command line for ON")
-        }else{
-                yourcommand_Args.sendCommand("Additional Argument to command line for OFF")
-        }
+rule "begin your execution"
+when
+   Item YourTrigger changed
+then
+      // here we can take different actions according to source Item
+   if(YourTrigger.state == ON){
+      yourcommand_Args.sendCommand("Additional Argument to command line for ON")
+   }else{
+      yourcommand_Args.sendCommand("Additional Argument to command line for OFF")
+   }
+
+      // Trigger execution (if autorun false)
+   yourcommand_Run.sendCommand(ON)
+      // the Run indicator state will go ON shortly, and return OFF when script finished
+end
 
-      // wait for the command to complete
-      // State will be NULL if not used before or ON while command is executed
-      while(yourcommand.state != OFF){
-         Thread::sleep(500)
-      }
-      
-      // Trigger execution
-      yourcommand.sendCommand(ON)
-      
-      // Logging of command line result
-      logInfo("Your command exec", "Result:" + yourcommand_out.state )
-      
+rule "process your results"
+when
+   Item yourcommand_Run changed from ON to OFF
```

Comment:
Would "Item yourcommand_Out updated" not be better? That is the item that is used in the rule.

Reply:
I'm sure either would work in practice.  Define "better" :)
As this is an example, using the Run channel demonstrates how that feature works.
In truth, I'm not sure if the Out channel does get updated if there is a script error, which might influence any user choice here.

Reply:
The openHAB event bus is async. I'm wondering if it would be possible that the yourcommand_Run already changed state from ON to OFF triggering the rule while the yourcommand_Out item has not been updated yet. Then you'd be logging and posting an old yourcommand_Out state. So in that sense I think using yourcommand_Out would be "better". However, for documentation purposes this is probably fine.

Reply:
I would hope that exec binding posts the Out content before changing the Run state, but I can't read the code, so it's only me hoping.  I realize that does not guarantee sequence, but by the time we've triggered the rule and examined the new Out state, all should be well?